ocean_provider/routes.py

```diff
@@ -273,6 +273,56 @@ def checkURL():
     )
 
 
+@services.route('/assetInfo', methods=['POST'])
+def assetInfo():
+    """Return info about each file in the asset (index, valid, contentLength, contentType)
+
+    This can be used by the publisher of an asset to check basic information
+    about the URL. For now, this information consists of the Content-Type
+    and Content-Length of the request, using primarily OPTIONS, with fallback
+    to GET. In the future, we will add a hash to make sure that the file was
+    not tampered with at consumption time.
+
+    tags:
+      - services
+
+    responses:
+      200:
+        description: the URL could be analysed (returns the result).
+      204:
+        description: the URL could be analysed, no data was found (no result).
+      400:
+        description: the URL could not be analysed (no result).
+
+    return: the URL's Content-type and ContentLength
+    """
+    required_attributes = ['documentId', ]
+    data = get_request_data(request)
+
+    msg, status = check_required_attributes(
+        required_attributes, data, 'assetInfo'
+    )
+    if msg:
+        return jsonify(error=msg), status
+
+    did = data['documentId']
+    asset = get_asset_from_metadatastore(get_metadata_url(), did)
+    url_list = get_asset_urls(asset, provider_wallet)
+    files_info = []
+    for i, url in enumerate(url_list):
+        download_url = get_download_url(url, app.config['CONFIG_FILE'])
```

Comment:
I'm not sure, but I believe if get_asset_urls is called before, we don't need to convert again with get_download_url. Maybe that's why the ipfs test is failing.  I'll check again tomorrow.

Reply:
correct, thanks for the change.